test/unittests/test_utils_test.cpp

```diff
@@ -159,6 +159,53 @@ TEST(test_utils, result_value_matcher)
     EXPECT_THAT(TypedExecutionResult(Value{1_u64}, ValType::i64), Not(Result(uint8_t{1})));
 }
 
+TEST(test_utils, cresult_value_matcher)
+{
+    using testing::Not;
+
+    FizzyExecutionResult trap{true, false, FizzyValue{}};
+    EXPECT_THAT(trap, CTraps());
+    EXPECT_THAT(trap, Not(CResult()));
+
+    FizzyExecutionResult result_void{false, false, FizzyValue{}};
+    EXPECT_THAT(result_void, CResult());
+    EXPECT_THAT(result_void, Not(CTraps()));
+
+    // Each result type is checked only against values with equal type, because for other types
+    // CResult matcher would access non-active union members of result's value, which presents UB.
+
+    EXPECT_THAT(trap, Not(CResult(0_u32)));
+    EXPECT_THAT(result_void, Not(CResult(0_u32)));
+    FizzyValue value_i32{0};
```

Comment:
Would we be better not using 0 but actual values in these tests?

Reply:
Not sure, but maybe we would, I'll change it.